state/store.go

```diff
@@ -25,16 +26,37 @@ const (
 
 //------------------------------------------------------------------------
 
-func calcValidatorsKey(height int64) []byte {
-	return []byte(fmt.Sprintf("validatorsKey:%v", height))
+const (
+	// prefixes are unique across all tm db's
+	prefixValidators      = byte(0x05)
+	prefixConsensusParams = byte(0x06)
+	prefixABCIResponses   = byte(0x07)
+)
+
+func encodeKey(buf []byte, height int64) []byte {
+	res, _ := orderedcode.Append(buf, height)
+	return res
+}
+
+func decodeKey(key []byte) int64 {
+	var height int64
+	_, err := orderedcode.Parse(string(key[1:]), &height)
```

Comment:
why [1:] ?

Reply:
The first byte is the prefix which is ignored

Reply:
```suggestion
	const prefixLen = 1
	_, err := orderedcode.Parse(string(key[prefixLen:]), &height)
```